azavea/fb-gender-survey-dashboard
# Add 2021 Data

## Overview

Processed and integrated the 2021 data into the app. A few minor
changes were required in the processing script related to dropped
columns, but in general processing the new data went as planned.

Connects #119 

### Demo

2020 Data
<img width="1283" alt="2020" src="https://user-images.githubusercontent.com/21046714/139307853-fc704720-f658-48bb-aba4-addd4fd9a3a5.png">

2021 Data
<img width="1241" alt="2021" src="https://user-images.githubusercontent.com/21046714/139307883-dd64c9a0-2917-4a9a-9f9f-9e44a48a9d0d.png">

### Notes

'Data unavailable' will still show for some selections, when viewing questions which have been weren't asked in the currently selected year. Selected these questions will be prevented in #113.

## Testing Instructions

 * Run `./scripts/server`
 * Select a country and 2020, then select some or all questions. Confirm that the charts render appropriately and do not break. 
 * Select a region and 2020, then select some or all questions. Confirm that the charts render appropriately and do not break. 
 * Select a country and 2021, then select some or all questions. Confirm that the charts render appropriately and do not break. 
 * Select a region and 2021, then select some or all questions. Confirm that the charts render appropriately and do not break. 
 * 'Save' a 2021 visualization page, then go to 'Saved Charts' and reload the chart. Confirm that it reloads with 2021 data.
